integration/skydb.test.ts

```diff
@@ -240,4 +246,190 @@ describe(`SkyDB end to end integration tests for portal '${portal}'`, () => {
 
     expect(data).toEqual(json);
   });
+
+  it("Should update the revision number cache", async () => {
+    const { publicKey, privateKey } = genKeyPairAndSeed();
+    const json = { message: 1 };
+
+    await client.db.setJSON(privateKey, dataKey, json);
+
+    const cachedRevisionEntry = await client.revisionNumberCache.getRevisionAndMutexForEntry(publicKey, dataKey);
+    expect(cachedRevisionEntry.revision.toString()).toEqual("0");
+
+    await client.db.setJSON(privateKey, dataKey, json);
+
+    expect(cachedRevisionEntry.revision.toString()).toEqual("1");
```

Comment:
Maybe it makes more sense for the `revisionNumberCache` to be a field in the `client.db`. e.g. `client.db.revisionNumberCache` instead. Then it makes sense that calling `db.setJSON` can update the cache because it essentially owns it. While now it appears to be this separate thing.